frame/contracts-registry/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+
+		/// Maximum length of contract data.
+		#[pallet::constant]
+		type MaxContractContentLen: Get<u32>;
+
+		/// Maximum signatures per contract
+		#[pallet::constant]
+		type MaxSignatures: Get<u32>;
+
+		/// Maximum judges count
+		#[pallet::constant]
+		type MaxJudges: Get<u32>;
+
+		/// Maximum parties per contract
+		#[pallet::constant]
+		type MaxContractParties: Get<u32>;
+
+		/// Origin from which a judge may be added
+		type AddJudgeOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// Origin from which the new typical action can be made.
+		///
+		/// The success variant is the account id of the depositor.
+		type SubmitOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::AccountId>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Maximum number of items reached.
+		TooMany,
+		/// Can not found contract at given id
+		ContractNotFound,
+		/// Contract is already signed by caller
+		AlreadySigned,
+		/// Caller isn't judge
+		NotJudge,
+		/// Caller isn't parties
+		NotParties,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A judge has been added.
+		AddedJudge { judge_id: ContractIndex, deposit: T::AccountId },
+		/// Judge signed a contract
+		JudgeSigned { contract_id: ContractIndex, signer: T::AccountId },
+		/// Judge signed a contract
+		PartieSigned { contract_id: ContractIndex, signer: T::AccountId },
+		/// Created new contract
+		ContractCreated { contract_id: ContractIndex, creator: T::AccountId },
+	}
+
+	/// Contracts storage containing a content of contracts
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn contracts)]
+	pub type Contracts<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		ContractsData<T::MaxContractContentLen, T::MaxContractParties, T::AccountId>,
+		OptionQuery,
+	>;
+
+	/// Vec of parties signatures for each contract
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn parties_signatures)]
+	pub type PartiesSignatures<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		BoundedVec<T::AccountId, T::MaxSignatures>,
+		OptionQuery,
+	>;
+
+	/// Vec of judge signatures for each contract
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn judge_signatures)]
+	pub(super) type JudgeSignatures<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		BoundedVec<T::AccountId, T::MaxSignatures>,
+		OptionQuery,
+	>;
+
+	/// List of judges
+	#[pallet::storage]
+	#[pallet::getter(fn judges)]
+	pub type Judges<T: Config> =
+		StorageValue<_, BoundedVec<T::AccountId, T::MaxJudges>, ValueQuery>;
+
+	/// The number of contracts that have been made so far.
+	#[pallet::storage]
+	#[pallet::getter(fn contracts_count)]
+	pub type ContractsCount<T> = StorageValue<_, ContractIndex, ValueQuery>;
```

Comment:
This is a bad name, as it will become confusing if we ever add deleting contracts.
Maybe track `NextContractId` instead?